spring-boot/proxy/src/main/java/eu/coatrack/proxy/metrics/MetricsCounterService.java

```diff
@@ -21,131 +21,95 @@
-                .append(SEPARATOR)
-                .append(httpResponseCode) // [5]
-                .append(SEPARATOR)
-                .append(today) // [6]
-                .append(SEPARATOR)
-                .append(path) // [7]
-                .toString());
     }
 
-    public Metric filterAndTransformMetric(org.springframework.boot.actuate.metrics.Metric inputMetric) {
-
-        String[] elements = inputMetric.getName().split(SEPARATOR);
-        Assert.noNullElements(elements);
-
-        // check if this is a custom metric
-        if (elements[0].equals("counter." + PREFIX)) {
-            log.debug("Metrics for transmission: " + inputMetric.getName());
+    private void incrementAndSetCount(Metric metricToTransmit) {
+        String counterId = createCounterIdFromMetric(metricToTransmit);
 
-            Metric outputMetric = new Metric();
-            outputMetric.setRequestMethod(elements[2]);
-            outputMetric.setCount(inputMetric.getValue().intValue());
-            outputMetric.setMetricsCounterSessionID(counterSessionID);
-            outputMetric.setType(MetricType.valueOf(elements[4]));
+        Counter counter = meterRegistry.find(counterId).counter();
+        if (counter == null){
+            counter = meterRegistry.counter(counterId);
+        }
+        counter.increment();
+        int currentCount = (int) counter.count();
+        metricToTransmit.setCount(currentCount);
+    }
 
-            // add a tmp api key object with just the key value, will later be mapped to entity on CoatRack admin side
-            ApiKey tmpApiKey = new ApiKey();
-            tmpApiKey.setKeyValue(elements[3]);
-            outputMetric.setApiKey(tmpApiKey);
+    private String createCounterIdFromMetric(Metric metric) {
+        StringJoiner stringJoiner = new StringJoiner(SEPARATOR);
+        stringJoiner.add(PREFIX)
+                .add(metric.getApiKey().getServiceApiName())
+                .add(metric.getRequestMethod())
+                .add(metric.getApiKey().getKeyValue())
+                .add(metric.getType().toString())
+                .add(String.valueOf(metric.getHttpResponseCode()))
+                .add(LocalDate.now().toString())
+                .add(metric.getPath());
+        return stringJoiner.toString();
+    }
 
-            String httpStatusString = elements[5];
-            Integer httpStatus = httpStatusString.equals("null") ? null : new Integer(httpStatusString);
-            outputMetric.setHttpResponseCode(httpStatus);
+    private Metric createMetricToTransmit(MetricsHolder mh) {
+        Metric metricToTransmit = new Metric();
 
-            String dateString = elements[6];
-            log.debug("date string is: {}", dateString);
-            LocalDate dateOfApiUsage = LocalDate.parse(dateString);
-            log.debug("date of api usage parsed: {}", dateOfApiUsage);
-            outputMetric.setDateOfApiCall(Date.valueOf(dateOfApiUsage));
+        ServiceApi serviceApi = new ServiceApi();
```

Comment:
The old code did not contain the creation of a `ServiceApi` entity. See old lines 127 to 130. Or did I miss it?

Anyway: This "dummy ServiceApi" object seems to lead to a NullPointerException when creating the JSON for transmission. So I would suggest to do it like it was done in the "old" code.

Reply:
You are right. I deleted the unnecessary lines for the service.